programs/mpl-core/src/plugins/oracle.rs

```diff
@@ -57,4 +77,89 @@ pub struct OracleUpdateInfo {
     pub lifecycle_checks: Option<Vec<(HookableLifecycleEvent, ExternalCheckResult)>>,
     /// Optional PDA (derived from Pubkey attached to `ExternalPluginKey`).
     pub pda: Option<ExtraAccount>,
+    /// Optional format for validation results struct used in Oracle account.  Default
+    /// is `ValidationResultsStructFormat::Standard`.
+    pub format: Option<ValidationResultsStructFormat>,
+    /// Optional offset for validation results struct used in Oracle account.  Default
+    /// is `ValidationResultsOffset::NoOffset`.
+    pub results_offset: Option<ValidationResultsOffset>,
+}
+
+/// Format of the validation results struct used in Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub enum ValidationResultsStructFormat {
+    /// `Standard` means the Oracle will contain the `OracleValidation` type.
+    Standard,
+    /// `Compact` means the Oracle will contain the `OracleValidationCompact` type.
+    Compact,
+}
+
+/// Offset to where the validation results struct is located in an Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub enum ValidationResultsOffset {
+    /// The validation struct is located at the beginning of the account.
+    NoOffset,
+    /// The Oracle is an Anchor account so the validation struct is located after an 8-byte
+    /// account discriminator.
+    Anchor,
+    /// The validation struct is located at the specified offset within the account.
+    Custom(usize),
+}
+
+/// Validation results struct for an Oracle account.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub struct OracleValidation {
```

Comment:
I think because this is data we can't control, we should use the old Args pattern where we have
```
pub enum OracleValidation {
    V1 {},
}
```
So that way we can deserialize the whole thing and adding new lifecycle events in the future is easy with Versioning.

Reply:
Oh yeah we talked about this but I forgot it.  Added

Reply:
Yeah the way i was accomplishing this was with the "standard" field, but I think the version on the account is better because it does not require the account to be in a matched state with the asset plugin.